pkg/dataobj/internal/sections/streams/streams.go

```diff
@@ -61,10 +77,26 @@ func New(metrics *Metrics, pageSize int) *Streams {
 	return &Streams{
 		metrics:  metrics,
 		pageSize: pageSize,
-		lookup:   make(map[uint64][]*Stream),
+		lookup:   make(map[uint64][]*Stream, 1024),
+		ordered:  make([]*Stream, 0, 1024),
 	}
 }
 
+func (s *Streams) Iter() result.Seq[Stream] {
+	return result.Iter(func(yield func(Stream) bool) error {
+		for _, stream := range s.ordered {
+			if !yield(*stream) {
+				return nil
+			}
+		}
+		return nil
+	})
+}
+
+func (s *Streams) GetBounds() (time.Time, time.Time) {
+	return s.globalMinTimestamp, s.globalMaxTimestamp
+}
```

Comment:
can we add doc comments to this?

Also maybe another name would be more descriptive here, like `TimeRange` or `CurrentRange`. Bounds makes me think that streams outside of the bounds are rejected, which isn't the case.